pkg/cmd/audit/audit_filter.go

```diff
@@ -222,13 +222,30 @@ func URIToParts(uri string) (string, schema.GroupVersionResource, string) {
 			return ns, gvr, name
 		}
 
-		if parts[2] != "namespaces" {
+		switch{
+		case parts[2] != "namespaces" : // cluster scoped request that is not a namespace
 			gvr.Resource = parts[2]
 			if len(parts) >= 4 {
 				name = parts[3]
 				return ns, gvr, name
 			}
+		case len(parts) == 3 && parts[2] == "namespaces" : // a namespace request /api/v1/namespaces
+			gvr.Resource = parts[2]
+			return "", gvr, ""
+
+		case len(parts) == 4 && parts[2] == "namespaces", // a namespace request /api/v1/namespaces/<name>
+			len(parts) == 5 && parts[2] == "namespaces" && parts[2] == "finalize", // a namespace request /api/v1/namespaces/<name>/finalize
+			len(parts) == 5 && parts[2] == "namespaces" && parts[2] == "finalize": // a namespace request /api/v1/namespaces/<name>/status
```

Comment:
These two lines look identical, although the comments are different. 

Furthermore, the "finalize" should be compared against parts[4], isn't it? 

From the comments it appears you probably want to do this:
```
len(parts) == 5 && parts[2] == "namespaces" && parts[4] == "finalize", // a namespace request /api/v1/namespaces/<name>/finalize
			len(parts) == 5 && parts[2] == "namespaces" && parts[4] == "status": // a namespace request /api/v1/namespaces/<name>/status 
```

Reply:
awkward. Thanks